app/models/changelog_entry.rb

```diff
@@ -36,9 +44,25 @@ def created_by?(user)
     created_by == user
   end
 
+  def tweet_link
+    case
+    when details_html? then friendly_url
+    when info_url? then info_url
+    else ""
+    end
+  end
+
+  def friendly_url
+    @friendly_url ||= FriendlyUrl.new(self).url
```

Comment:
I feel like we could maybe just simplify this to `Rails.application.routes.changelog_url(to_param)` or something, rather than having the extra class? Thoughts?

For this you could use `FriendlyId` which does all of this work for you (and is in the codebase)and then not need `custom url_slug` etc. I don't massively mind either way as long as we have test-coverage, but is there any reason you preferred this?

Reply:
@iHiD For Friendly ID, it saves a slug to the database when the record is created. With how we've set up the slug right now, we won't be able to use it since we depend on the record's ID. We could do it as a column update after the record is saved, but figured that it might be inefficient. I've read articles that recommend using the Friendly ID's candidates instead, but we need to work out what candidates to use. Happy to change this up to use friendly ID but we'd need to sort out how to generate slugs for these better.

I agree with the suggestion of simplifying this to not use the extra class. It's a lot simpler.

Reply:
Great. That's cool. Let's just go with this for now then. I think it'll work great.